crates/sui-framework/sources/transfer.move

```diff
@@ -13,7 +14,14 @@ module sui::transfer {
     /// Transfer ownership of `obj` to `recipient`. `obj` must have the
     /// `key` attribute, which (in turn) ensures that `obj` has a globally
     /// unique ID.
-    public fun transfer<T: key>(obj: T, recipient: address) {
+    public fun transfer<T: key>(obj: T, recipient: Recipient) {
+        transfer_internal(obj, sui::address::from_recipient(recipient))
+    }
+
+    /// Transfer ownership of `obj` to an address `recipient`. `obj` must have the
+    /// `key` attribute, which (in turn) ensures that `obj` has a globally
+    /// unique ID.
+    public fun transfer_to_address<T: key>(obj: T, recipient: address) {
```

Comment:
What's the advice we give people on when to use this function directly vs supporting `transfer` on `Recipient`?  I'm not 100% clear on that from the changes in this PR.